reana/reana_dev/release.py

```diff
@@ -190,6 +190,12 @@ def release_pypi(ctx, component, timeout):  # noqa: D301
 
 
 @click.option("--user", "-u", default="reanahub", help="DockerHub user name [reanahub]")
+@click.option(
+    "--maint",
```

Comment:
Instead of calling the new option `--maint`, I would call it `--commit` (to be conform to `cr` option name) or `--cr-commit` (to be very explicit how it is used) with the help message saying something like "specify target commit for release? [default=no]" and explain in detail in the docstring that `--cr-commit` is to be used only when releasing non-master branches such as `maint-0.9` or `next`.

(Ideally, we should be able to always use this also for master branch, so that there would be no need for such a flag, but this solution would be as an MVP.)